### PR TITLE
Add DataCite.xml metadata

### PR DESCRIPTION
**Pull request by max-zilla**
This is the first step in adoption RDA recommendations as described here: 
https://opensource.ncsa.illinois.edu/confluence/display/CATS/Adoption+of+RDA+Interoperability+Recommendations

This adds the datacite.xml output, although there are a few questions remaining.
- still need to add best way of getting Clowder instance URL for external access.
- Is our handling of rights sufficient for now?
- Do we need to track Family and Given name separately in users? DataCite specifies Family, Given as format but in Clowder we normally use Given Family and supporting that may require guesswork or new fields in Dataset model.
- Is the idea of contributors adequately addressed (file & metadata uploaders in the dataset)?

Example output:
```
<resource xsi:schemaLocation="http://datacite.org/schema/kernel-4 http://schema.datacite.org/meta/kernel-4/metadata.xsd">
<identifier identifierType="DOI">:none</identifier>
<creators>
	<creator>
		<creatorName>Max Burnette</creatorName>
		<nameIdentifier>0000-0002-1786-3676</nameIdentifier>
		<nameIdentifierScheme>ORCID</nameIdentifierScheme>
	</creator>
</creators>
<titles>
	<title>Test Dataset</title>
</titles>
<publisher>Clowder</publisher>
<publicationYear>2020</publicationYear>
<resourceType resourceTypeGeneral="Dataset">Clowder Dataset</ResourceType>
<descriptions>
	<description descriptionType="Abstract">this can be used for all kinds of stuff.</description>
</descriptions>
<dates>
	<date dateType="Created">2020-06-30</date>
</dates>
<alternateIdentifier>5efb8514066cdd1a900c7a69</alternateIdentifier>
<format>application/zip</format>
<subjects>
	<tag>Just Making Sure</tag>
</subjects>
<relatedIdentifier relatedIdentifierType="URL" relationType="isSourceOf">CLOWDER URL GOES HERE</ResourceType>
<sizes>
	<size>4 files</size>
</sizes>
<rights rightsURI="http://creativecommons.org/licenses/by-nc-nd/3.0/">Attribution-NonCommercial-NoDerivs</rights>
</resource>
```

### CODE REVIEW

**Comment by max-zilla** [comment]
Discovered a bug with the tagmanifest file - will update when fixed.

**Comment by max-zilla** [comment]
OK, fixed 2 bugs: 1) bagit MD5 was getting written to the data MD5 file and not the bag file, this was a bug I introduced 2) the manifest MD5 file was writing a _metadata.json entry even though the file is _dataset_metadata.json and can be in the /data folder - fixed logic so it is indexed properly.

**Comment by tcnichol** [comment]
Looks good as far as I can tell. Tested the bagit downloads for datasets and collections and all seems to be in order.

File: app/api/Datasets.scala [hunk 1]
```diff
@@ -2408,6 +2359,122 @@ class  Datasets @Inject()(
...
+
+    // Subjects
+    if (dataset.tags.length > 0) {
+      s += "<subjects>\n"
+      dataset.tags.foreach(t => s += "\t<tag>"+t.name+"</tag>\n")
+      s += "</subjects>\n"
+    }
+
+    // RelatedIdentifier
+    val clowder_url = "CLOWDER URL GOES HERE"
```
Comment (lmarini) [inline app/api/Datasets.scala:2455]: What do you plan on doing for this url? Should we put the actual url before we merge this PR?

Comment (lmarini) [reply app/api/Datasets.scala:2455]: I think we talked about it but I completely forgot! :(

Comment (max-zilla) [reply app/api/Datasets.scala]: Now the original requests that call this enumerator will pass along the URL from their request so the dataset link will be included there.

**Pull request merged**